src/util.rs

```diff
@@ -53,6 +61,16 @@ pub(crate) trait BufExt2: Buf {
         }
         return unsafe { CString::from_vec_unchecked(bytes) };
     }
+
+    fn get_tlv_map<T : FromPrimitive + Eq + Hash>(&mut self) -> HashMap<T, Vec<u8>> {
+        let mut parameters = HashMap::new();
+        while self.has_remaining() {
+            let parameter_type: T = self.get_primitive_u16_le();
+            let value_size = self.get_u8() as usize;
+            parameters.insert(parameter_type, self.get_vec_u8(value_size));
+        }
+        parameters
+    }
```

Comment:
This is a handy function, but the name doesn't make it immediately obvious what it does. Can you add a `/// comment` please?

Reply:
Done.